arbitrary-users-patch/build_images.sh

```diff
@@ -31,7 +33,11 @@ while read -r line; do
   base_image_name=$(echo "$line" | tr -s ' ' | cut -f 1 -d ' ')
   base_image=$(echo "$line" | tr -s ' ' | cut -f 2 -d ' ')
   echo "Building ${NAME_FORMAT}/${base_image_name}:${TAG} based on $base_image ..."
-  docker build -t "${NAME_FORMAT}/${base_image_name}:${TAG}" --no-cache --build-arg FROM_IMAGE="$base_image" "${SCRIPT_DIR}"/
+  script_dir=${SCRIPT_DIR}
+  if [[ ${base_image:0:5} = ${MAVEN_BASE} ]] ; then
```

Comment:
This seems like a too-restrictive check. What if the base image name is just a java image which happens to have maven in it? https://github.com/redhat-developer/codeready-workspaces/blob/master/dependencies/che-devfile-registry/devfiles/02_java-maven/devfile.yaml#L18

Why not do a check like this?

```
  # substring check for "java" or "maven" in ${base_image}
  if [[ ${base_image} = *"maven"* ]] || [[ ${base_image} = *"java"* ]]; then 
```

Reply:
I continue to be strongly against this sort of patch in the arbitrary user patch section. If we want an image with `MAVEN_CONFIG` unset then we should build one explicitly.

Reply:
Also worth noting that this workaround is *specific* to the [dockerhub community maven image](https://github.com/carlossg/docker-maven/blob/master/jdk-8/mvn-entrypoint.sh#L45), and is not necessary in e.g. an image where you install maven yourself.